### PR TITLE
[SR SQL Planner Summer Camp] Support anltr4 for ShowProcesslistStmt

### LINKED ISSUE #7960: New Parser and Analyzer support 19. ShowProcesslistStmt （Easy）
# Required work and output

1.  Use antlr4 to complete parse statements. The generated statement must follow the original data structure, and should be compatible with the old statement.
    
2.  In the original syntax file  **sql_parser.cup**, the new syntax needs to be compatible with the syntax of the old version, using the syntax file sql_parser.cup of the old version as the reference standard.
    
3.  Remove the logic of analytic functions in the statement and move the relevant logic to the new Analyzer framework. The new Analyzer is in the SQL/Analyzer directory.
    
4.  If the code related to Privilege Check exists in the original analyzer, move the code related to privilege control into PrivilegeChecker.
    
5.  When submitting the PR, please make sure there is unit test coverage for related changes.
    

# Task
 - [ ] ShowProcesslistStmt

# How to participate?

1.  Pick an issue you are interested in, click the issue link and leave a comment under the issue to sign up.  **One person can only choose one issue and the first person who picks the issue will be assigned.**
    
2.  **After the issue is assigned, you should complete the task in 14 days.** If you are unable to complete it within the specified time, the task will be withdrawn and assigned to others.
    
3.  Participants are required to produce two pieces of content (forms including but not limited to mind maps, study notes, blogs, and tutorials) in the course of the activity, each of which should be at least 200 words.
    
4.  Got questions? Join our  [Slack channel](https://join.slack.com/t/starrocks/shared_invite/zt-192zeqlc7-9MAvqAXNcveW_l5Pqr8r2g)  and find channel:  **#sql-planner**.
    

# Before you start

1.  Read  [Contributing.md](https://github.com/StarRocks/starrocks/blob/main/CONTRIBUTING.md)  carefully to understand how to submit a pull request and how to configure BE and FE development environment.
    
2.  Sign the  [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks)  before you make a pull request.
    
3.  To learn more about StarRocks, please refer to  [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).
    

# Pull request requirements

1.  Create a pull request (PR) after you complete the task.
* PR title format: [SR SQL Planner Summer Camp] + PR title
2. After submitting PR, you can comment under the pinned Issue [SR SQL Planner Summer Camp].
* Comment format: "I have finished Issue #" + PR link

### PR DESCRIPTION
**Pull request by predator4ann**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：
Fixes #7960 

## Problem Summary(Required) ：
Support anltr4 for ShowProcesslistStmt

### CODE REVIEW

**Comment by predator4ann** [comment]
run starrocks_fe_unittest

**Comment by predator4ann** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// This file is made available under Elastic License 2.0.
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java:1]: new files should use starrocks license header
// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.

**Comment by imay** [comment]
@predator4ann you can rebase main branch to make fe unittest run success

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java (hunk 1 above)
Comment (predator4ann) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java]: done

**Comment by predator4ann** [comment]
> @predator4ann you can rebase main branch to make fe unittest run success

@imay I rebase main branch already and fe unittest still failed

**Comment by predator4ann** [comment]
@imay thanks for reviewing, I have found the failed reason that there are some fixed time("storage_cooldown_time", "2022-07-09 12:12:12") in unittest

**Comment by predator4ann** [comment]
@hiliuxg

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcesslistStmt.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+// This file is based on code available under the Apache license here:
```
Comment (HangyuanLiu) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java:2]: delete Apache License

Comment (predator4ann) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcesslistStmtTest.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcesslistStmt.java|3|4|75.00%|[63]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|

**Review by HangyuanLiu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/ShowProcesslistStmt.java|3|4|75.00%|[63]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|

**Pull request closed without merging**